wallet/src/main/java/io/neow3j/wallet/Wallet.java

```diff
@@ -142,6 +186,11 @@ public boolean addAccount(Account account) {
      * @return true if an account was removed, false if no account with the given address was found.
      */
     public boolean removeAccount(ScriptHash scriptHash) {
+        if (scriptHash == this.getDefaultAccount().getScriptHash()
```

Comment:
Use `.equals()` instead of `==`. The `scriptHash` argument might not be the same object that is referenced in the default account. In such a scenario the boolean expression would evaluate to false, but should.